addons/html_editor/static/src/others/x2many_media_viewer.js

```diff
@@ -0,0 +1,99 @@
+import { registry } from '@web/core/registry';
+import { useService } from '@web/core/utils/hooks';
+import { X2ManyField, x2ManyField } from '@web/views/fields/x2many/x2many_field';
+import { MediaDialog } from '../main/media/media_dialog';
+import { getVideoUrl } from '../utils/url';
+
+export class X2ManyMediaDialog extends MediaDialog {
```

Comment:
> What does this class bring that the MediaDialog doesn't? And can't that be included in the MediaDialog instead?

This override brings an added functionality to the `save` method. Original code simply prepares the attachments and runs `props.save` but with IMG elements which won't work in our case especially with record updating and caching. Hence this override is placed. We may include it in MediaDialog but then it may hinder with website's functionality hence I kept it separate.

> From reading the code there looks to be something about some "preloaded attachments" but there is nothing about that in the spec so I don't know what this is about.

Basically media dialog allows addition of images from unsplash and URLs along with user uploaded images. There is one edge case for which I have to introduce `preloaded attachments`. Let's say I selected couple of preloaded images in media dialog, and then searched for something in unsplash and selected that too. Now my save method will run directly with that data, for the preloaded images it will find its records in `ir.attachment` but for unsplash images, it first needs to run the original save in order to download and save it in attachments followed by running my set of code for adding it in product images.

> Also, the class is inside the x2many_media_viewer.js file, which feels wrong.

I was a bit skeptical about that since it was just a save override, I will split it into a separate file.

Reply:
What do you mean by "it may hinder with website's functionality"? What would happen if we had this code in website linked to the `multiImages` option for example?

About the case of unsplash and preloaded attachments, can't we already have that issue now with `multiImages`? What happens currently in a media dialog with `multiImages = true`? Can't I select a preloaded image and an unsplash image at the same time?

Reply:
Current `MediaDialog` is programmed in such a way that it modifies DOM elements on a html field upon adding images. In this particular use case, we are using it to store image records, hence a complete modification of save method was done. 

Alternatively, we can modify it in generic MediaDialog, by adding a new flag that separates it's `save` logic from DOM elements to storing data in records. But that's up to a whole new discussion. What's your opinion on that?